frontend/src/component/Activity/MemberActivity.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import { translate } from 'react-i18next'
+import { escape as escapeHtml } from 'lodash'
+
+import {
+  Avatar,
+  DistanceDate,
+  TLM_ENTITY_TYPE as TLM_ET,
+  TLM_CORE_EVENT_TYPE as TLM_CET,
+  ROLE_LIST,
+  SUBSCRIPTION_TYPE_LIST
+} from 'tracim_frontend_lib'
+
+require('./MemberActivity.styl')
+
+export class MemberActivity extends React.Component {
+  getSpaceMemberText (coreEventType, i18nOpts) {
+    const { props } = this
+    const role = ROLE_LIST.find(r => r.slug === props.activity.newestMessage.fields.member.role)
+
+    i18nOpts = {
+      ...i18nOpts,
+      role: props.t(role.label)
+    }
+    switch (coreEventType) {
+      case TLM_CET.CREATED:
+        return props.t('{{user}} joined the space', i18nOpts)
+      case TLM_CET.MODIFIED:
+        return props.t("{{user}}'s role has been changed to {{role}}", i18nOpts)
+    }
+  }
+
+  getSpaceSubscriptionText (coreEventType, i18nOpts) {
+    const { props } = this
+    const subscription = SUBSCRIPTION_TYPE_LIST.find(s => s.slug === props.activity.newestMessage.fields.subscription.state)
+    i18nOpts = {
+      ...i18nOpts,
+      state: props.t(subscription.label)
+    }
+    switch (coreEventType) {
+      case TLM_CET.CREATED:
+        return props.t('{{user}} wants to join the space', i18nOpts)
+      case TLM_CET.MODIFIED:
+        return props.t("{{user}}'s request to join the space has been {{state}}", i18nOpts)
```

Comment:
We will only have activities for when the request is rejected, to avoid double activities (request accepted + became a member)

Reply:
Yes, this case is now handled in `ActivityFeed.jsx`.